### PR TITLE
UIREQ-945 create Jest/RTL test for NoteViewRoute.js

### PR DESCRIPTION
**Pull request by cchitneedi**
Refs:
UIREQ-945 create Jest/RTL test for NoteViewRoute.js

URL: https://issues.folio.org/browse/UIREQ-945

### CODE REVIEW

File: test/jest/__mock__/stripesSmartComponents.mock.js [hunk 1]
```diff
@@ -4,6 +4,31 @@ jest.mock('@folio/stripes/smart-components', () => ({
...
+    );
+  },
+  NoteEditPage: (props) => {
+    return (
+      <div>
+        <div>NoteEditPage</div>
+        {props.renderReferredRecord()}
+      </div>
+    );
+  },
```
Comment (artem-blazhko) [inline test/jest/__mock__/stripesSmartComponents.mock.js:22]: I do not see that NoteCreatePage and NoteEditPage are used inside `NoteViewRoute` component.
Could you remove them?

File: test/jest/__mock__/stripesSmartComponents.mock.js [hunk 2]
```diff
@@ -4,6 +4,31 @@ jest.mock('@folio/stripes/smart-components', () => ({
...
+  NoteViewPage: (props) => {
+    return (
+      <div>
+        <div>NoteViewPage</div>
+        {props.renderReferredRecord()}
+        <button type="button" onClick={() => props.onEdit()}>onEdit</button>
```
Comment (artem-blazhko) [inline test/jest/__mock__/stripesSmartComponents.mock.js:28]:
```suggestion
        <button type="button" onClick={props.onEdit}>onEdit</button>
```

File: src/routes/NoteViewRoute.test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+describe('NoteViewRoute', () => {
+  it('NoteViewPage should render when location.state is not empty', () => {
+    renderNoteViewRoute(locationData, historyData);
+    expect(screen.getByText('NoteViewPage')).toBeInTheDocument();
+  });
+  it('history.replace function to be called when onEdit clicked', () => {
+    renderNoteViewRoute(locationData, historyData);
+    userEvent.click(screen.getByRole('button', { name: 'onEdit' }));
+    expect(historyData.replace).toBeCalled();
+  });
```
Comment (artem-blazhko) [inline src/routes/NoteViewRoute.test.js:33]: Could you move these tests in separate `describe` and call `renderNoteViewRoute` only once inside `beforeEach`
```suggestion
describe('when "location.state" is not empty', () => {
  beforeEach(() => {
      renderNoteViewRoute(locationData, historyData);
  });

   it('should render "NoteViewPage"', () => {
      expect(screen.getByText('NoteViewPage')).toBeInTheDocument();
    });
    
    it('should trigger "history.replace"', () => {
      userEvent.click(screen.getByRole('button', { name: 'onEdit' }));
      
      expect(historyData.replace).toBeCalled();
    });
});
```

**Review by artem-blazhko — CHANGES_REQUESTED** [review]

File: test/jest/__mock__/stripesSmartComponents.mock.js (hunk 1 above)
Comment (cchitneedi) [reply test/jest/__mock__/stripesSmartComponents.mock.js]: Changed done

File: src/routes/NoteViewRoute.test.js (hunk 3 above)
Comment (cchitneedi) [reply src/routes/NoteViewRoute.test.js]: Changes done in test file

**Review by Dmitriy-Litvinenko — APPROVED** [review]
Hello. Thank you. Could you please update or rebase branch and merge it?

**Pull request merged**